app.js

```diff
@@ -1,97 +1,140 @@
+        bio: userData.bio,
+        avatar: userData.avatar_url,
+        html_url: userData.html_url,
+      };
+    }));
+  } catch (error) {
+    console.error("Error fetching authors:", error);
+    return [];
   }
-  return authors;
 }
 
+// set authors in the DOM
 async function setAuthors(apiURL) {
-  // Getting authors and placing them
-  const authors = await getAuthors(apiURL);
-  for (const author of authors) {
-    document.querySelector("#creators > div > div").innerHTML += `
-    <div class="person">
-      <div class="name_and_img">
-        <img src="${author.avatar}" alt="">
-        <p class="person_name">${author.name}</p>
-      </div>
-      <p class="about_person">${author.bio}</p>
-     </div>
-     `;
+  try {
+    const authors = await getAuthors(apiURL);
+    const container = document.querySelector("#creators > div > div");
+    if (!container) {
+      throw new Error("Container element not found");
+    }
+    authors.forEach(author => {
+      const personDiv = document.createElement('div');
+      personDiv.className = 'person';
+      personDiv.innerHTML = `
+                <div class="name_and_img">
+                    <img src="${author.avatar}" alt="${author.name}'s avatar">
+                    <p class="person_name">${author.name}</p>
+                </div>
+                <p class="about_person">${author.bio || 'No biography available'}</p>
+            `;
+      container.appendChild(personDiv);
+    });
+  } catch (error) {
+    console.error("Error setting authors:", error);
   }
 }
 
+// set contributors in the DOM
 async function setContributors(apiURL, element) {
-  const contributors = await getAuthors(apiURL);
-  if (contributors.length === 0) {
-    element.innerHTML = "No developers found";
-    return;
-  }
-  
-  // Clear loading message
-  element.innerHTML = "";
-  // List developer names
-  for (const contributor of contributors) {
-    element.innerHTML += `
-      <span class="developer-name">
-        <a href="${contributor.html_url}" class="tooltip" target="_blank">  
-          <img src="${contributor.avatar}" alt="" class="shining-image">  
-          <span class="tooltip-text">${contributor.name}</span>  
-        </a>
-      </span>
-    `;
+  try {
+    const contributors = await getAuthors(apiURL);
+    element.innerHTML = contributors.length === 0 ? "No developers found" :
+      contributors.map(contributor => `
+                <span class="developer-name">
+                    <a href="${contributor.html_url}" class="tooltip" target="_blank">  
+                        <img src="${contributor.avatar}" alt="${contributor.name}'s avatar" class="shining-image">  
+                        <span class="tooltip-text">${contributor.name}</span>  
+                    </a>
+                </span>
+            `).join('');
+  } catch (error) {
+    console.error("Error setting contributors:", error);
+    element.innerHTML = "Error loading developers";
```

Comment:
this is good. Thank you 🙌 

let's make the message more user-friendly

```suggestion
    element.innerHTML = "Can't load the developer list. Check out the repository page to see all of the contributors 😅";
```